external-crates/move/testing-infra/transactional-test-runner/src/framework.rs

```diff
@@ -512,16 +523,18 @@ pub fn compile_any<'a, A: MoveTestAdapter<'a>>(
     command_lines_stop: usize,
     _stop_line: usize,
     data: Option<NamedTempFile>,
-    handler: impl FnOnce(
-        &mut A,
-        Vec<(Option<Symbol>, CompiledModule)>,
-    ) -> Result<(Option<String>, Vec<(Option<Symbol>, CompiledModule)>)>,
+    handler: F,
 ) -> Result<(
     Option<String>,
     Option<String>,
     NamedTempFile,
     Vec<(Option<Symbol>, CompiledModule)>,
-)> {
+)>
+where
+    A: MoveTestAdapter<'a> + 'b,
+    F: FnOnce(&'b mut A, Vec<(Option<Symbol>, CompiledModule)>) -> R,
+    R: Future<Output = Result<(Option<String>, Vec<(Option<Symbol>, CompiledModule)>)>> + 'c,
```

Comment:
I had quite a difficult time getting this to compile. And my lifetime soup here makes rust happy, but the rust analyzer dies

Reply:
Some more descriptive names for the lifetime parameters would be helpful here: 

- `'a = 'deps`, 
- `'b = 'adapter`,
- `'c = 'compilation`,

?